### ISSUE TITLE
Upgrade pnpm to v7

### ISSUE THREAD

**Issue by wenche** [equinor/energyvision#1051]
pnpm v7 is released, and we should look into [what it takes to upgrade](https://github.com/pnpm/pnpm/releases/tag/v7.0.0)

**Comment by fernandolucchesi** [comment]
dependent on #1040

**Comment by wenche** [comment]
@storybook/addon-docs uses v1.x.x of @mdx-js/react and that version does not support React 18 (version 2 does) See also https://github.com/storybookjs/storybook/issues/18383 for some related discussions. I don't think this is of great importance for us since we are not using mdx

**Comment by wenche** [comment]
@storybook/addon-essentials -> @storybook/addon-actions -> react-inspector [does not yet support React 18](https://github.com/storybookjs/react-inspector/issues/152)

**Comment by wenche** [comment]
```
┬ @equinor/eds-core-react
│ └─┬ @equinor/eds-utils
│   └─┬ react-popper
│     └── ✕ unmet peer react@"^16.8.0 || ^17": found 18.1.0
```
react-popper v2.30 has fixed this, but the EDS team are holding back the upgrade for React 18 a bit. We don't use any components from EDS that relies on react-popper, so this is not a big deal for us

**Comment by wenche** [comment]
[Forgot to add the one for Reach](https://github.com/reach/reach-ui/issues/916)

No progress on any of the issues yet.

**Comment by nilsml** [comment]
Why isn't this closed? pnpm is using latest version already

**Comment by wenche** [comment]
Because we are waiting on updates for all the library mentioned in the comments to get rid of peerDeps warnings

**Comment by SvSven** [comment]
No official React 18 support for reach-ui yet

**Comment by millianapia** [comment]
as we are phasing out reach-ui #1378, we can start looking at this again

**Comment by fernandolucchesi** [comment]
Ready to be worked on after #1532

**Comment by SvSven** [comment]
According to [the migration guide for v8](https://github.com/pnpm/pnpm/releases/tag/v8.0.0), we need to regenerate the pnpm lockfile as they have upgraded to lockfile version 6, that's missing in the PR still. Should be as simple as making sure your local pnpm version is 8.5.1 similar to the ones pinned in the PR, and then running `pnpm setup-project` - it'll autoregenerate the lockfiles then - it did for me at least.

Haven't fully tested the web and studio with the new lockfile/pnpm v8 yet though

Also looks like there isn't a Dockerfile in the v3 studio yet, but we'll need that to deploy the satellite studios as well. We can take that in another issue

**Comment by SvSven** [comment]
Have pushed updated lock files, we should make sure everything still works as expected 😬

**Comment by millianapia** [comment]
i have checked that the web and studio still works as expected, all of the regular pnpm commands still works. Other things to have in mind before i merge @fernandolucchesi @padms @SvSven ?

**Comment by fernandolucchesi** [comment]
and dockerfile of sanityv3 should also be updated

**Comment by fernandolucchesi** [comment]
reviewed sanity v3 docker and works fine! ✅
